src/track/track.cpp

```diff
@@ -900,26 +916,104 @@ void Track::setCuePoints(const QList<CuePointer>& cuePoints) {
             cuePoints);
 }
 
-Track::CueImportStatus Track::importCueInfos(
+Track::ImportStatus Track::importBeats(
+        mixxx::BeatsImporterPointer pBeatsImporter) {
+    QMutexLocker lock(&m_qMutex);
+    VERIFY_OR_DEBUG_ASSERT(pBeatsImporter) {
+        return ImportStatus::Complete;
+    }
+    DEBUG_ASSERT(!m_pBeatsImporterPending);
+    m_pBeatsImporterPending = pBeatsImporter;
+    if (m_streamInfo) {
+        // Replace existing cue points with imported cue
+        // points immediately
+        importPendingBeatsMarkDirtyAndUnlock(&lock);
+        return ImportStatus::Complete;
+    } else {
+        if (m_pBeatsImporterPending->isEmpty()) {
+            // Just return the current import status without clearing any
+            // existing cue points.
+            return (!m_pBeatsImporterPending->isEmpty())
+                    ? ImportStatus::Complete
+                    : ImportStatus::Pending;
+        }
```

Comment:
This is basically copy/pasted from https://github.com/mixxxdj/mixxx/pull/2958/files#diff-f59a0f669757d180c782c7a8fa0adea0R1011-R1017.

@uklotzde Shouldn't this always return `Complete`? Right now I think it always returns `Pending`, right?

Reply:
Oh, this check redundant. Yes, please fix it and return Complete when empty. Might be a remainder from a previous refactoring.